frontend/src/features/admin-form/create/builder-and-design/utils/questionNumbers.ts

```diff
@@ -0,0 +1,16 @@
+import { FormFieldDto } from '~shared/types/field'
+
+import { BASICFIELD_TO_DRAWER_META } from '../../constants'
+
+export const getBuilderQuestionNumbers = (
+  fields: FormFieldDto[],
+): (string | undefined)[] => {
+  let questionCounter = 0
+  return fields.map((field) => {
+    if (BASICFIELD_TO_DRAWER_META[field.fieldType].isSubmitted) {
+      questionCounter += 1
+      return `${questionCounter}.`
```

Comment:
I dunno if it's too late to give this comment, but I'd recommend keeping this as number (i.e. meaningful data), and do the eventual transformation to string later, as close to the point of rendering as possible. That would leave the actual renderer in control of how to display, whether with a dot-suffix, or paren-suffix, or whatever else the renderer wants to do.

Also super nit 😅 
```javascript
return ++questionCounter;
```

Reply:
made the `++` change in [f3d04a0](https://github.com/opengovsg/FormSG/pull/3692/commits/f3d04a09ae7d5be1c86cfda056a1259064cb8629), but would you be okay with keeping the data flow as-is for now? I'm going to refactor towards the approach in #3625 once it's merged anyway, and modifying the data flow will cause merge conflicts

Reply:
Yup, yup, let's go!